packages/private/edtr-io/src/plugins/solution-steps/index.ts

```diff
@@ -0,0 +1,99 @@
+import {
+  child,
+  object,
+  list,
+  string,
+  boolean,
+  EditorPluginProps,
+  EditorPlugin
+} from '@edtr-io/plugin'
+
+import { SolutionStepsEditor } from './editor'
+import * as Guidelines from './helper/guideline-texts'
+
+function createSolutionStepState(
+  introduction: Parameters<typeof child>,
+  strategy: Parameters<typeof child>,
+  solutionStep: Parameters<typeof child>,
+  additionals: Parameters<typeof child>
+) {
+  const solutionStepState = object({
+    type: string('step'),
+    isHalf: boolean(),
+    content: child(...solutionStep)
+  })
+  return object({
+    introduction: child(...introduction),
+    strategy: child(...strategy),
+    hasStrategy: boolean(),
+    solutionSteps: list(solutionStepState),
+    additionals: child(...additionals),
+    hasAdditionals: boolean()
+  })
+}
+
+export type SolutionStepsState = ReturnType<typeof createSolutionStepState>
+export type SolutionStepsProps = EditorPluginProps<SolutionStepsState, Config>
+interface Component {
+  options: Parameters<typeof child>
+  guideline: React.ReactNode
+  placeholder: string
+}
+export interface Config {
+  introduction: Component
+  strategy: Component
+  step: Component
+  explanation: Component
+  additionals: Component
+}
+
+export function createSolutionStepsPlugin({
```

Comment:
We don't need a config here since we are only using it for serlo anyways

Reply:
done